tests/test_shadow_fork.py

```
@@ -73,31 +73,53 @@ def test_create_fails_if_already_active(self, tmp_path):
         """Test that create warns about uncommitted changes."""
         fork = ShadowFork(repo_root=tmp_path)
 
-        with patch.object(fork, "_run_git") as mock_git:
-            mock_git.side_effect = [
-                (True, "main", ""),  # Get current branch
-                (True, " M file.txt", ""),  # Uncommitted changes exist
-                (True, "", ""),  # Verify base branch exists
-                (True, "", ""),  # Create and switch to shadow branch
-            ]
-
+        def mock_git_side_effect(cmd, **kwargs):
```

Ruff (0.15.5)

[warning] Missing return type annotation for private function `mock_git_side_effect`

(ANN202)